packages/enzyme-adapter-react-helper/src/getAdapterForReactVersion.js

```diff
@@ -1,30 +1,30 @@
 import semver from 'semver';
 
 export default function getAdapterForReactVersion(reactVersion) {
-  const normalizedVersion = semver.coerce(reactVersion);
+  const versionRange = semver.validRange(reactVersion);
 
-  if (semver.satisfies(normalizedVersion, '^16.4.0-0')) {
+  if (semver.intersects(versionRange, '^16.4.0')) {
```

Comment:
Hey @ljharb , I know you said before that handling pre-release versions was important. Unfortunately, I can't figure out a good solution here that handles pre-releases and matches versions appropriately.

What's the use case for handling pre-releases here? I may be missing something, but in my mind since users of this package provide a specific react version (through either a CLI arg or an env variable), they can provide a general version that makes sense for them (ie - 16.5). Even if they're on a pre-release version of React.

Reply:
anybody still depending on a prerelease of react 16 - which includes a number of our packages, including enzyme itself - would be unable to use this helper without that support.

In other words, they'd need to be able to provide a `REACT` value that explicitly installed a prerelease.

Reply:
Okay, so what you're saying is we need to be able to do something like:

```
REACT=16.8.6-apha.666
```

Which would definitely break as a result of this PR. I've been searching all morning and have not found a solution that gives us that while also giving us the correct version for non-pre-release versions. Was hopeful that semver's `includePrelease` option would help, but it doesn't seem to.

Only thing I can think of now is to detect the prelease and have a different set of conditionals for those 😑

Reply:
or `REACT=16.9.0-alpha` etc.